src/gsi/setupps.f90

```diff
@@ -321,7 +328,10 @@ subroutine setupps(obsLL,odiagLL,lunin,mype,bwork,awork,nele,nobs,is,conv_diagsa
      ioff0=20
      nreal=ioff0
      if (lobsdiagsave) nreal=nreal+4*miter+1
-     if (twodvar_regional) then; nreal=nreal+2; allocate(cprvstg(nobs),csprvstg(nobs)); endif
+     if (twodvar_regional .or. l_rtma3d .or. l_obsprvdiag) then
```

Comment:
Line 331 only needs `if (l_obsprvdiag) then`

As discussed before, let "l_obsprvdiag" solely decide what to do, not if it is 2D/3D RTMA.  If RTMA wants to output subprovider information, just set "l_obsprvdiag" to "true" in the namelist. 

Suggest to remove all these kind of unnecessary if (twodvar_regional .or. l_rtma3d in other places.

Also note that "l_rtma3d" is previously added for ghost reflectivity removal in the cloud analysis part. If a RTMA version does not run cloud analysis, using "if l_rtma3d" will be problematic.

Reply:
I can remove the trigger of "l_rtma3d" in setup?.f90 to control the output of obs provider info. 
twodvar_regional might still be necessary for backward compatible to operational (2D)RTMA sono need to modify the script and namelist file to add the "l_obsprvdiag". Anyway, I will discuss with Manuel and Matthew for this.